src/test/java/no/training/project/VersionTest.java

```diff
@@ -0,0 +1,120 @@
+package no.training.project;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import no.training.project.exception.ServiceException;
+import no.training.project.model.ReleaseResponse;
+import no.training.project.resource.version.VersionResource;
+import org.junit.jupiter.api.Assertions;
+import org.mockito.Mock;
+import java.io.IOException;
+import java.net.http.HttpClient;
+import java.net.http.HttpResponse;
+import java.util.List;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+import org.junit.jupiter.api.Test;
+
+public class VersionTest {
+    @Mock
+    private HttpClient defaultHttpClient;
+    @Mock
+    private ObjectMapper objectMapper;
+
+
+    private static ReleaseResponse getRelease() throws IOException {
+        String response = new String(VersionTest.class.getResourceAsStream("/mock-data/release-response.json").readAllBytes());
+        ReleaseResponse responseAsObject = new ObjectMapper().readValue(response, ReleaseResponse.class);
+        return responseAsObject;
+    }
+
+    @Test
+    public void successTest() throws IOException, InterruptedException, ServiceException {
+        HttpClient httpClient = mock(HttpClient.class);
+        HttpResponse response = mock(HttpResponse.class);
+        when(response.statusCode()).thenReturn(200);
+        String externalResponseString = new String(VersionTest.class.getResourceAsStream("/mock-data/release-response.json").readAllBytes());
+        when(response.body()).thenReturn(externalResponseString);
+
+        HttpResponse response2= mock(HttpResponse.class);
+        when(response2.statusCode()).thenReturn(404);
+        VersionResource version = new VersionResource(httpClient);
+        when(httpClient.send(any(), any())).thenReturn(response, response2);
+        List<String> release = version.getReleaseVersion("arm");
+        Assertions.assertNotNull(release);
+    }
+
+
+    @Test
+    public void validAndNonParsableResponse() throws IOException, InterruptedException, ServiceException {
```

Comment:
In this test, it's quite important to check if we throw an exception at all. Now, if there is no exception, `verify` calls in the `catch` block will be silently ignored, and the test will pass. 

Let's check for an exception this way: 
```
        assertThrows(RuntimeException.class, () → {
            // Original test code without try-catch
        });

```